src/components/app.js

```diff
@@ -244,10 +259,17 @@ const App = () => {
                       (c) => c.process === matchProcess.params.processName
                   )
                 : -1;
-        index !== -1
-            ? setSelectedTabId(matchProcess.params.processName)
-            : setSelectedTabId(false);
-    }, [configs, matchProcess]);
+        if (index !== -1) {
+            setSelectedTabId(matchProcess.params.processName);
+        } else {
+            if (configs.length > 0) {
+                setSelectedTabId(configs[0].process);
+                history.replace('/processes/' + configs[0].process);
```

Comment:
Why history.replace only in this case ?

Reply:
Then this code should not be here, it should be in the handler of the button to delete the process.

Also did you use replace on purpose vs push ?